internal/benchmarks/benchmarks_test.go

```diff
@@ -263,13 +263,15 @@ func BenchmarkAllGadgetsWithContainers(b *testing.B) {
 							operatorsParamCollection,
 							parser,
 							logger.DefaultLogger(),
+							0,
 						)
 
 						_, err := runtime.RunGadget(gadgetCtx)
 						if err != nil {
 							b.Fatalf("running gadget: %s", err)
 						}
 
+						gadgetCtx.Cancel()
```

Comment:
This seems weird to cancel two contexts there. Can't we rely on the parent being cancelled in the line below?

Reply:
Mmmm, yes, we can, but is it a good practice to always call the `cancel()`?

Reply:
I wonder if we should support this zero timeout in the gadget-context. We could pass `timeout=-1` by default when it needs to wait forever, and any other value (including `0`) will be respected.